### PR TITLE
Add support of the SA/PA facilities.

### PR DESCRIPTION
**Pull request by MaximAlien**
PR adds support of such SA/PA facilities: 
- `Amenity`
- `AmenityType`

### CODE REVIEW

File: Sources/MapboxDirections/RestStop.swift [hunk 1]
```diff
@@ -7,11 +7,17 @@ import Turf
 public struct RestStop: Codable, Equatable, ForeignMemberContainer {
     public var foreignMembers: JSONObject = [:]
 
-    /// A kind of rest stop.
+    /**
```
Comment (MaximAlien) [inline Sources/MapboxDirections/RestStop.swift:10]: Just aligns comments to use similar style in this file.

**Review by jill-cardamon — APPROVED** [review]
lgtm besides the comments I left!

File: Sources/MapboxDirections/Amenity.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        type = try container.decode(AmenityType.self, forKey: .type)
+        name = try container.decodeIfPresent(String.self, forKey: .name)
+        brand = try container.decodeIfPresent(String.self, forKey: .brand)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(type, forKey: .type)
+        try container.encode(name, forKey: .name)
+        try container.encode(brand, forKey: .brand)
```
[the remark this answers is not in the document]
Comment (MaximAlien) [reply Sources/MapboxDirections/Amenity.swift:53]: Good catch. I think yes, will make the change.

File: Sources/MapboxDirections/RestStop.swift [hunk 3]
```diff
@@ -36,23 +50,27 @@ public struct RestStop: Codable, Equatable, ForeignMemberContainer {
...
     
     /**
      Initializes an optionally named rest stop of a certain kind.
      
      - parameter type: The kind of rest stop.
      - parameter name: The name of the rest stop.
+     - parameter amenities: Facilities associated with the rest stop.
      */
-    public init(type: StopType, name: String?) {
+    public init(type: StopType, name: String?, amenities: [Amenity]? = nil) {
```
[the remark this answers is not in the document]
Comment (MaximAlien) [reply Sources/MapboxDirections/RestStop.swift:63]: I don't think this default parameter leads to a breaking change. I've also integrated current implementation into https://github.com/mapbox/mapbox-navigation-ios/pull/4300 and there are no build issues.

[the remark this answers is not in the document]
Comment (MaximAlien) [reply Sources/MapboxDirections/RestStop.swift:65]: I can see that it was detected during API diff step:
```
1 breaking change detected in MapboxDirections:
  💔 API breakage: constructor RestStop.init(type:name:) has been removed
```
At the same time I still feel that it's safe to provide default value in existing initializer.

**Pull request merged**